### PR TITLE
add debug string to query plan [MINOR]

### PR DESCRIPTION
**Pull request by agavra**
Adds functionality to be able to call `QueryPlan#explain()` to see an `EXPLAIN`-like string. Here are some examples:

`SELECT * FROM b ORDER BY col1, col2 DESC LIMIT 3`
```
[0] MAIL_RECEIVE
└── [1] MAIL_SEND
    └── [1] SORT (LIMIT 3)
        └── [1] MAIL_RECEIVE
            └── [2] MAIL_SEND
                └── [2] TABLE SCAN (b) {Server_localhost_56777={REALTIME=[b1]}}
```

`SELECT a.col1, a.col3, b.col3 FROM a JOIN b ON MOD(a.col3, 2) = MOD(b.col3, 3)`
```
[0] MAIL_RECEIVE
└── [1] MAIL_SEND
    └── [1] PROJECT
        └── [1] JOIN
            ├── [1] MAIL_RECEIVE
            │   └── [2] MAIL_SEND
            │       └── [2] PROJECT
            │           └── [2] TABLE SCAN (a) {Server_localhost_56778={REALTIME=[a3]}, Server_localhost_56777={REALTIME=[a1, a2]}}
            └── [1] MAIL_RECEIVE
                └── [3] MAIL_SEND
                    └── [3] PROJECT
                        └── [3] TABLE SCAN (b) {Server_localhost_56777={REALTIME=[b1]}}
```

`SELECT col1, SUM(col3) FROM a WHERE a.col3 BETWEEN 23 AND 36 GROUP BY col1 HAVING SUM(col3) > 10.0 AND MIN(col3) <> 123 AND MAX(col3) BETWEEN 10 AND 20`
```
[0] MAIL_RECEIVE
└── [1] MAIL_SEND
    └── [1] PROJECT
        └── [1] FILTER
            └── [1] AGGREGATE
                └── [1] MAIL_RECEIVE
                    └── [2] MAIL_SEND
                        └── [2] AGGREGATE
                            └── [2] FILTER
                                └── [2] TABLE SCAN (a) {Server_localhost_56778={REALTIME=[a3]}, Server_localhost_56777={REALTIME=[a1, a2]}}
```

### CODE REVIEW

File: pinot-query-planner/src/main/java/org/apache/pinot/query/planner/stage/AggregateNode.java [hunk 1]
```diff
@@ -54,4 +54,9 @@ public List<RexExpression> getAggCalls() {
   public List<RexExpression> getGroupSet() {
     return _groupSet;
   }
+
+  @Override
+  public String debugString() {
```
Comment (siddharthteotia) [inline pinot-query-planner/src/main/java/org/apache/pinot/query/planner/stage/AggregateNode.java:59]: In the existing engine, we implement `toExplainString` at the physical operator level and each operator also has a EXPLAIN_NAME. 

Even in this case, I think the API could be called `explainString` or something similar even though we are doing at the logical node level and not at the operator level.

**Review by siddharthteotia — APPROVED** [review]
Consider not calling the API `debugString`. 

LGTM otherwise

File: pinot-query-planner/src/main/java/org/apache/pinot/query/planner/stage/AggregateNode.java (hunk 1 above)
Comment (siddharthteotia) [reply pinot-query-planner/src/main/java/org/apache/pinot/query/planner/stage/AggregateNode.java:59]: Ohh I just noticed this is not wired into the EXPLAIN PLAN. We do support EXPLAIN for multi-stage btw. We can probably also do this by wiring this up with EXPLAIN PLAN for multi-stage (can do as a follow up as well).

 The output in the description looks reasonably good

Comment (siddharthteotia) [reply pinot-query-planner/src/main/java/org/apache/pinot/query/planner/stage/AggregateNode.java:59]: can we also add instances / worker assignment into the output ?

Comment (agavra) [reply pinot-query-planner/src/main/java/org/apache/pinot/query/planner/stage/AggregateNode.java]: @siddharthteotia you read my mind 😉 that's _just_ what I was working on (I'll update PR description with examples once I've worked out the bugs).

**Review by siddharthteotia — APPROVED** [review]

**Comment by siddharthteotia** [comment]
We should consider doing a follow-up to wire this into EXPLAIN - https://github.com/apache/pinot/pull/9280

**Pull request merged**